cupy/random/generator.py

```diff
@@ -90,17 +90,24 @@ def binomial(self, n, p, size=None, dtype=int):
             self.rk_seed += numpy.prod(size)
         return y
 
+    _laplace_kernel = core.ElementwiseKernel(
+        'T x, T loc, T scale', 'T y',
+        'y = T(loc) + T(scale) * ((x < 0.5) ? log(x + x): -log(2.0 - x - x))',
+        'laplace_kernel')
+
     def laplace(self, loc=0.0, scale=1.0, size=None, dtype=float):
         """Returns an array of samples drawn from the laplace distribution.
 
         .. seealso::
             :func:`cupy.random.laplace` for full documentation,
             :meth:`numpy.random.RandomState.laplace`
         """
-        x = self.uniform(size=size, dtype=dtype)
-        loc = cupy.asarray(loc, dtype)
-        scale = cupy.asarray(scale, dtype)
-        _kernels.laplace_kernel(x, loc, scale, x)
+        x = self.random_sample(size=size, dtype=dtype)
+        if not numpy.isscalar(loc):
```

Comment:
Why do this?
When `loc` is scalar, `loc` remain scalar.
When `loc` is array, `loc` remain array.

Reply:
The elementwise kernel can take scalar value and `cupy.ndarray`.

Reply:
OK.

```
if not numpy.isscalar(loc):
    loc = cupy.asarray(loc, dtype)
```
The reason that the code is necessary is to equalize dtype? or to support `numpy.ndarray`?

Reply:
The kernel requires same dtype array.
The scalar values are casted automatically.

Reply:
OK.